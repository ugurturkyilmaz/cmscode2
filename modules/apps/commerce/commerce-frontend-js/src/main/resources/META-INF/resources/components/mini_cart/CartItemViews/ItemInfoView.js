/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

import PropTypes from 'prop-types';
import React from 'react';

function ItemInfoViewOptions({options}) {
	return (
		<div className="item-info-extra mt-3">
			<h6 className="options">{options}</h6>
		</div>
	);
}

function ItemInfoViewBundle({childItems}) {
	return (
		<div className="child-items">
			{childItems.map((item, index) => {
				const {name, quantity} = item;

				return (
					<div className="child-item" key={index}>
						<span>
							{quantity} &times; {name}
						</span>
					</div>
				);
			})}
		</div>
	);
}

function ItemInfoViewBase({name, sku}) {
	return (
		<div className="item-info-base">
			<h5 className="item-name">{name}</h5>

			<p className="item-sku">{sku}</p>
		</div>
	);
}

function ItemInfoView({childItems = [], name, options = '', sku}) {
	const isBundle = childItems.length > 0;
	const hasOptions = !!options;

	return (
		<>
			<ItemInfoViewBase name={name} sku={sku} />

			{isBundle && <ItemInfoViewBundle childItems={childItems} />}

			{hasOptions && <ItemInfoViewOptions options={options} />}
		</>
	);
}

ItemInfoView.propTypes = {
	childItems: PropTypes.array,
	name: PropTypes.string.isRequired,
	options: PropTypes.string,
	sku: PropTypes.string.isRequired,
};

export default ItemInfoView;
