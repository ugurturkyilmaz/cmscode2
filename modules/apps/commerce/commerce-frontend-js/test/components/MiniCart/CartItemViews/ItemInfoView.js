/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

import '@testing-library/jest-dom/extend-expect';
import {cleanup, render} from '@testing-library/react';
import React from 'react';

import ItemInfoView from '../../../../src/main/resources/META-INF/resources/components/mini_cart/CartItemViews/ItemInfoView';

describe('MiniCart Item Info View', () => {
	const BASE_PROPS = {
		name: 'An Item',
		sku: 'ITEM001',
	};

	const COMPONENT_SELECTOR_BASE = '.item-info-base';
	const COMPONENT_SELECTOR_BUNDLE = '.child-items';
	const COMPONENT_SELECTOR_OPTIONS = '.item-info-extra';

	afterEach(() => {
		jest.resetAllMocks();

		cleanup();
	});

	describe('Base', () => {
		it('renders the name and the SKU of an item', () => {
			const {container} = render(<ItemInfoView {...BASE_PROPS} />);

			const BaseViewElement = container.querySelector(
				COMPONENT_SELECTOR_BASE
			);
			const ItemNameElement = BaseViewElement.querySelector('.item-name');
			const ItemSKUElement = BaseViewElement.querySelector('.item-sku');

			expect(BaseViewElement).toBeInTheDocument();
			expect(ItemNameElement).toBeInTheDocument();
			expect(ItemSKUElement).toBeInTheDocument();

			expect(ItemNameElement.innerHTML).toEqual(BASE_PROPS.name);
			expect(ItemSKUElement.innerHTML).toEqual(BASE_PROPS.sku);
			expect(BaseViewElement.innerHTML).toMatchSnapshot();
		});

		describe('+ Bundle', () => {
			const BUNDLE_PROPS = {
				...BASE_PROPS,
				childItems: [
					{
						name: 'Child Item 1',
						quantity: 1,
					},
					{
						name: 'Child Item 2',
						quantity: 3,
					},
				],
			};

			it('renders the name and the SKU of a bundle item, plus its child items with their related quantities', () => {
				const {asFragment, container} = render(
					<ItemInfoView {...BUNDLE_PROPS} />
				);

				const BaseViewElement = container.querySelector(
					COMPONENT_SELECTOR_BASE
				);
				const ItemNameElement = BaseViewElement.querySelector(
					'.item-name'
				);
				const ItemSKUElement = BaseViewElement.querySelector(
					'.item-sku'
				);

				expect(BaseViewElement).toBeInTheDocument();
				expect(ItemNameElement).toBeInTheDocument();
				expect(ItemSKUElement).toBeInTheDocument();

				expect(ItemNameElement.innerHTML).toEqual(BASE_PROPS.name);
				expect(ItemSKUElement.innerHTML).toEqual(BASE_PROPS.sku);

				const BundleViewElement = container.querySelector(
					COMPONENT_SELECTOR_BUNDLE
				);
				const ChildItemsElements = BundleViewElement.querySelectorAll(
					'.child-item span'
				);

				expect(BundleViewElement).toBeInTheDocument();
				expect(ChildItemsElements.length).toEqual(2);

				Array.from(ChildItemsElements).forEach((element) => {
					expect(element.innerHTML).toMatchSnapshot();
				});

				expect(asFragment()).toMatchSnapshot();
			});
		});

		describe('+ Options', () => {
			const OPTIONS_PROPS = {
				...BASE_PROPS,
				options: 'XL',
			};

			it('renders the name and the SKU of an item, plus its options', () => {
				const {asFragment, container} = render(
					<ItemInfoView {...OPTIONS_PROPS} />
				);

				const BaseViewElement = container.querySelector(
					COMPONENT_SELECTOR_BASE
				);
				const ItemNameElement = BaseViewElement.querySelector(
					'.item-name'
				);
				const ItemSKUElement = BaseViewElement.querySelector(
					'.item-sku'
				);

				expect(BaseViewElement).toBeInTheDocument();
				expect(ItemNameElement).toBeInTheDocument();
				expect(ItemSKUElement).toBeInTheDocument();

				expect(ItemNameElement.innerHTML).toEqual(BASE_PROPS.name);
				expect(ItemSKUElement.innerHTML).toEqual(BASE_PROPS.sku);

				const OptionsViewElement = container.querySelector(
					COMPONENT_SELECTOR_OPTIONS
				);
				const OptionsText = OptionsViewElement.querySelector(
					'.options'
				);

				expect(OptionsViewElement).toBeInTheDocument();
				expect(OptionsText.innerHTML).toEqual(OPTIONS_PROPS.options);

				expect(asFragment()).toMatchSnapshot();
			});
		});

		describe('+ Bundle + Options', () => {
			const BUNDLE_OPTIONS_PROPS = {
				...BASE_PROPS,
				childItems: [
					{
						name: 'Child Item 1',
						quantity: 1,
					},
					{
						name: 'Child Item 2',
						quantity: 3,
					},
					{
						name: 'Child Item 3',
						quantity: 5,
					},
				],
				options: 'XL',
			};

			it('renders the name and the SKU of an item, plus its child items with their related quantities, plus its options', () => {
				const {asFragment, container} = render(
					<ItemInfoView {...BUNDLE_OPTIONS_PROPS} />
				);

				const BaseViewElement = container.querySelector(
					COMPONENT_SELECTOR_BASE
				);
				const ItemNameElement = BaseViewElement.querySelector(
					'.item-name'
				);
				const ItemSKUElement = BaseViewElement.querySelector(
					'.item-sku'
				);

				expect(BaseViewElement).toBeInTheDocument();
				expect(ItemNameElement).toBeInTheDocument();
				expect(ItemSKUElement).toBeInTheDocument();

				expect(ItemNameElement.innerHTML).toEqual(BASE_PROPS.name);
				expect(ItemSKUElement.innerHTML).toEqual(BASE_PROPS.sku);

				const BundleViewElement = container.querySelector(
					COMPONENT_SELECTOR_BUNDLE
				);
				const ChildItemsElements = BundleViewElement.querySelectorAll(
					'.child-item span'
				);

				expect(BundleViewElement).toBeInTheDocument();
				expect(ChildItemsElements.length).toEqual(3);

				Array.from(ChildItemsElements).forEach((element) => {
					expect(element.innerHTML).toMatchSnapshot();
				});

				const OptionsViewElement = container.querySelector(
					COMPONENT_SELECTOR_OPTIONS
				);
				const OptionsText = OptionsViewElement.querySelector(
					'.options'
				);

				expect(OptionsViewElement).toBeInTheDocument();
				expect(OptionsText.innerHTML).toEqual(
					BUNDLE_OPTIONS_PROPS.options
				);

				expect(asFragment()).toMatchSnapshot();
			});
		});
	});
});
