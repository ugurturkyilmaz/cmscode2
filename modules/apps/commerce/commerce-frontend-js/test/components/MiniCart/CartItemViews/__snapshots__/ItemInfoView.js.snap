// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MiniCart Item Info View Base + Bundle + Options renders the name and the SKU of an item, plus its child items with their related quantities, plus its options 1`] = `"1 × Child Item 1"`;

exports[`MiniCart Item Info View Base + Bundle + Options renders the name and the SKU of an item, plus its child items with their related quantities, plus its options 2`] = `"3 × Child Item 2"`;

exports[`MiniCart Item Info View Base + Bundle + Options renders the name and the SKU of an item, plus its child items with their related quantities, plus its options 3`] = `"5 × Child Item 3"`;

exports[`MiniCart Item Info View Base + Bundle + Options renders the name and the SKU of an item, plus its child items with their related quantities, plus its options 4`] = `
<DocumentFragment>
  <div
    class="item-info-base"
  >
    <h5
      class="item-name"
    >
      An Item
    </h5>
    <p
      class="item-sku"
    >
      ITEM001
    </p>
  </div>
  <div
    class="child-items"
  >
    <div
      class="child-item"
    >
      <span>
        1 × Child Item 1
      </span>
    </div>
    <div
      class="child-item"
    >
      <span>
        3 × Child Item 2
      </span>
    </div>
    <div
      class="child-item"
    >
      <span>
        5 × Child Item 3
      </span>
    </div>
  </div>
  <div
    class="item-info-extra mt-3"
  >
    <h6
      class="options"
    >
      XL
    </h6>
  </div>
</DocumentFragment>
`;

exports[`MiniCart Item Info View Base + Bundle renders the name and the SKU of a bundle item, plus its child items with their related quantities 1`] = `"1 × Child Item 1"`;

exports[`MiniCart Item Info View Base + Bundle renders the name and the SKU of a bundle item, plus its child items with their related quantities 2`] = `"3 × Child Item 2"`;

exports[`MiniCart Item Info View Base + Bundle renders the name and the SKU of a bundle item, plus its child items with their related quantities 3`] = `
<DocumentFragment>
  <div
    class="item-info-base"
  >
    <h5
      class="item-name"
    >
      An Item
    </h5>
    <p
      class="item-sku"
    >
      ITEM001
    </p>
  </div>
  <div
    class="child-items"
  >
    <div
      class="child-item"
    >
      <span>
        1 × Child Item 1
      </span>
    </div>
    <div
      class="child-item"
    >
      <span>
        3 × Child Item 2
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`MiniCart Item Info View Base + Options renders the name and the SKU of an item, plus its options 1`] = `
<DocumentFragment>
  <div
    class="item-info-base"
  >
    <h5
      class="item-name"
    >
      An Item
    </h5>
    <p
      class="item-sku"
    >
      ITEM001
    </p>
  </div>
  <div
    class="item-info-extra mt-3"
  >
    <h6
      class="options"
    >
      XL
    </h6>
  </div>
</DocumentFragment>
`;

exports[`MiniCart Item Info View Base renders the name and the SKU of an item 1`] = `"<h5 class=\\"item-name\\">An Item</h5><p class=\\"item-sku\\">ITEM001</p>"`;
